did/builder.go

```diff
@@ -0,0 +1,157 @@
+package did
+
+import (
+	"reflect"
+
+	"github.com/TBD54566975/ssi-sdk/util"
+	"github.com/google/uuid"
+	"github.com/pkg/errors"
+)
+
+type DIDDocumentBuilder struct {
+	contexts []string
+	types    []string
+	*DIDDocument
+}
+
+const (
+	DIDDocumentLDContext string = "https://w3id.org/did/v1"
+	DIDDocumentType      string = "DIDDocument"
+	BuilderEmptyError    string = "builder cannot be empty"
+)
+
+// Create a new DID Document Builder
+func NewDIDDocumentBuilder() DIDDocumentBuilder {
+	contexts := []string{DIDDocumentLDContext}
+	types := []string{DIDDocumentType}
+	return DIDDocumentBuilder{
+		contexts: contexts,
+		types:    types,
+		DIDDocument: &DIDDocument{
+			ID:      uuid.NewString(),
+			Context: contexts,
+		},
+	}
+}
+
+// Builds the DID Document
+func (builder *DIDDocumentBuilder) Build() (*DIDDocument, error) {
+	if builder.IsEmpty() {
+		return nil, errors.New(BuilderEmptyError)
+	}
+
+	if err := builder.DIDDocument.IsValid(); err != nil {
+		return nil, errors.Wrap(err, "did doc not ready to be built")
+	}
+
+	return builder.DIDDocument, nil
+}
+
+func (builder *DIDDocumentBuilder) IsEmpty() bool {
+	if builder == nil || builder.DIDDocument == nil {
+		return true
+	}
+	return reflect.DeepEqual(builder, &DIDDocumentBuilder{})
+}
+
+func (builder *DIDDocumentBuilder) AddContext(context interface{}) error {
+	if builder.IsEmpty() {
+		return errors.New(BuilderEmptyError)
+	}
+	res, err := util.InterfaceToStrings(context)
+	if err != nil {
+		return errors.Wrap(err, "malformed context")
+	}
+	uniqueContexts := util.MergeUniqueValues(builder.contexts, res)
+	builder.contexts = uniqueContexts
```

Comment:
great question. I used https://github.com/TBD54566975/ssi-sdk/blob/main/credential/builder.go as a reference. The reason it was suggested was the following: 

On : https://github.com/TBD54566975/ssi-sdk/blob/main/credential/builder.go#L25 	
// contexts and types are kept to avoid having cast to/from interface{} values

I made an assumption here since the context is also the same data type, this is a reasonable implementation to rely upon for did document builder.